plugins/tracing/http.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2017 Caicloud Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tracing
+
+import (
+	"net/http"
+	"net/http/httptrace"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	tlog "github.com/opentracing/opentracing-go/log"
+)
+
+// Transport wraps a RoundTripper. If a request is being traced with
+// Tracer, Transport will inject the current span into the headers,
+// and set HTTP related tags on the span.
+type Transport struct {
+	// The actual RoundTripper to use for the request. A nil
+	// RoundTripper defaults to http.DefaultTransport.
+	http.RoundTripper
+}
+
+// RoundTrip implements the RoundTripper interface.
+func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
+	rt := t.RoundTripper
+	if rt == nil {
+		rt = http.DefaultTransport
+	}
+
+	span, ctx := StartSpanFromContext(req.Context(), "HTTP: "+req.Method)
```

Comment:
`StartSpanFromContext` will return nil if req.Context() doesn't have span, what if I want to start a new span using the transport?

Reply:
The operation name `"HTTP: "+req.Method` doesn't make any sense.

Reply:
If you want this span to be more meaningful, you should create a span of yourself.

Reply:
Maybe we should use `HTTP: / API/v1 / applications / : app`  as the operation name?

Reply:
You can have users pass operation name as part of transport construction, and use `HTTP: /api/v1/applications/:app` as a default fallback if user doesn't pass any.

Using `/api/v1/applications/:app` is better since i'm able to tell who is sending the request.  `HTTP: GET`, `HTTP: POST` is REALLY non-sense.

Reply:
Let me think